pkg/build/strategies/sti/postexecutorstep.go

```diff
@@ -79,20 +81,18 @@ func (step *removePreviousImageStep) execute(ctx *postExecutorStepContext) error
 	if step.builder.incremental && step.builder.config.RemovePreviousImage {
 		glog.V(3).Info("Executing step: remove previous image")
 		step.removePreviousImage(ctx.previousImageID)
-	} else {
-		glog.V(3).Info("Skipping step: remove previous image")
+		return nil
 	}
+	glog.V(3).Info("Skipping step: remove previous image")
 	return nil
 }
 
 func (step *removePreviousImageStep) removePreviousImage(previousImageID string) {
-	if previousImageID == "" {
-		return
-	}
-
-	glog.V(1).Infof("Removing previously-tagged image %s", previousImageID)
-	if err := step.docker.RemoveImage(previousImageID); err != nil {
-		glog.V(0).Infof("error: Unable to remove previous image: %v", err)
+	if previousImageID != "" {
+		glog.V(1).Infof("Removing previously-tagged image %s", previousImageID)
+		if err := step.docker.RemoveImage(previousImageID); err != nil {
+			glog.V(0).Infof("error: Unable to remove previous image: %v", err)
+		}
```

Comment:
This change looks like in the opposite direction. Normally we remove nesting, not add. The previous code seemed fine, return earlier under some condition, do something else in the general case.